### PR TITLE
[OptionList] Add verticalAlign prop to center option content

### LINKED ISSUE #3690: [OptionList] Using Thumbnail as media does not align the content to center
## Feature request summary

When passing a `Thumbnail` as media (shown as a prefix), the text that follows is not aligned to the center.

If this is the expected behaviours then the suggestion would be to add an additional prop to force-center alignement within the `OptionItem`

If this is not the expected behaviour then directly change the files. 

## Screenshot

<details>
<summary> Current Implementation </summary>

image

</details>

<details>
<summary> Requested Implementation </summary>

image

</details>

### PR DESCRIPTION
**Pull request by chanonroy**
### WHY are these changes introduced?

Fixes #3690 
- https://github.com/Shopify/polaris-react/issues/3690

### WHAT is this pull request doing?

- Adds the ability to specify a vertical alignment prop on the `OptionList` component.
- Currently, the content is always aligned to the top, which may be visually undesirable.
- This PR gives developers the ability to vertically align option content how they wish.

Screen Shot 2021-12-06 at 6 55 20 PM

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [x] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

**Comment by ghost** [comment]
👋 Thanks for opening your first pull request. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md).

File: src/components/OptionList/components/Option/Option.scss [hunk 1]
```diff
@@ -91,3 +91,15 @@ $control-vertical-adjustment: rem(2px);
...
+  align-items: flex-start;
+}
+
+.VerticalAlign--Center {
+  align-items: center;
+}
+
+.VerticalAlign--Bottom {
+  align-items: flex-end;
+}
```
Comment (husshegz) [inline src/components/OptionList/components/Option/Option.scss]: I looked at other instances where a variation pattern is used and this repo doesn't use the `{ClassName}--{Variation}` but rather `{ClassName}{Variation}`.

I suggest to change it for consistency.

Comment (husshegz) [reply src/components/OptionList/components/Option/Option.scss]: (I personally do prefer the `--` separation though)

**Review by husshegz — APPROVED** [review]
Have one small comment about the naming around the css classes. Other than that, the code checks out ✅ and I did 🎩 in the story and was able to align the items accordingly

<details>

<summary> 🎩  of the alignment </summary>


https://user-images.githubusercontent.com/23648577/145420066-00f3091b-93d0-4d42-adf7-607469535908.mp4


</details>

Question for repo-owners:

This is a small 💅 cosmetic change to the component, so it usually wouldn't need it's own story, but i noticed that we're missing a story to show `media` prop. Maybe we can take this opportunity and add a story to showcase both `Media` and `Alignment`.

File: src/components/OptionList/components/Option/Option.scss (hunk 1 above)
Comment (chloerice) [reply src/components/OptionList/components/Option/Option.scss]: There's magic under the hood that autogenerates BEM classes in the final CSS bundled from SCSS. For prop/JS controlled variations, we use their camelcase names (e.g., for prop `verticalAlign` we'd make the class name `verticalAlign{Center/Top/Bottom}` and the output classname in the browser in dev mode would be something like `Polaris__Option--verticalAlignCenter`), you can checkout [the Stack stylesheet](https://github.com/Shopify/polaris-react/blob/b95fe67e5d33ddd271ce9729415750616bc1858d/src/components/Stack/Stack.scss#L28) for more examples of that.

```suggestion
.verticalAlignTop {
  align-items: flex-start;
}

.verticalAlignCenter {
  align-items: center;
}

.verticalAlignBottom {
  align-items: flex-end;
}
```

File: src/components/OptionList/components/Option/Option.tsx [hunk 2]
```diff
@@ -59,13 +63,15 @@ export function Option({
     disabled && styles.disabled,
     select && styles.select,
     active && styles.active,
+    verticalAlign && styles[variationName('VerticalAlign--', verticalAlign)],
```
Comment (chloerice) [inline src/components/OptionList/components/Option/Option.tsx]:
```suggestion
    verticalAlign && styles[variationName('verticalAlign', verticalAlign)],
```

File: UNRELEASED.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
 - Removed animtion from `Skeleton` components ([#4697](https://github.com/Shopify/polaris-react/pull/4697))
 - Remove duplicate duration(fast) usage. ([#4682](https://github.com/Shopify/polaris-react/pull/4682))
 - Updated the accessability label for the rollup actions in the `Page` header ([#4080](https://github.com/Shopify/polaris-react/pull/4080))
+- Add `verticalAlign` prop to OptionList. ([#4800](https://github.com/Shopify/polaris-react/pull/4800))
```
Comment (chloerice) [inline UNRELEASED.md]:
```suggestion
- Added a `verticalAlign` prop to `OptionList` ([#4800](https://github.com/Shopify/polaris-react/pull/4800))
```

File: src/components/OptionList/OptionList.tsx [hunk 4]
```diff
@@ -30,6 +32,8 @@ export interface OptionListProps {
   selected: string[];
   /** Allow more than one option to be selected */
   allowMultiple?: boolean;
+  /** Adjust vertical alignment of elements */
```
Comment (chloerice) [inline src/components/OptionList/OptionList.tsx]:
```suggestion
  /** Vertically align child content to the center, top, or bottom.  */
```

File: src/components/OptionList/components/Option/tests/Option.test.tsx [hunk 5]
```diff
@@ -114,6 +114,46 @@ describe('<Option />', () => {
...
+      });
+    });
+
+    it('renders with bottom alignment', () => {
+      const option = mountWithApp(
+        <Option {...defaultProps} allowMultiple verticalAlign="bottom" />,
+      );
+
+      expect(option).toContainReactComponent('label', {
+        className: 'Label VerticalAlign--Bottom',
```
Comment (chloerice) [inline src/components/OptionList/components/Option/tests/Option.test.tsx]:
```suggestion
        className: 'Label verticalAlignBottom',
```

File: src/components/OptionList/components/Option/tests/Option.test.tsx [hunk 6]
```diff
@@ -114,6 +114,46 @@ describe('<Option />', () => {
...
+      });
+    });
+
+    it('renders with center alignment', () => {
+      const option = mountWithApp(
+        <Option {...defaultProps} allowMultiple verticalAlign="center" />,
+      );
+
+      expect(option).toContainReactComponent('label', {
+        className: 'Label VerticalAlign--Center',
```
Comment (chloerice) [inline src/components/OptionList/components/Option/tests/Option.test.tsx]:
```suggestion
        className: 'Label verticalAlignCenter',
```

File: src/components/OptionList/components/Option/tests/Option.test.tsx [hunk 7]
```diff
@@ -114,6 +114,46 @@ describe('<Option />', () => {
...
+      });
+    });
+
+    it('renders with top alignment', () => {
+      const option = mountWithApp(
+        <Option {...defaultProps} allowMultiple verticalAlign="top" />,
+      );
+
+      expect(option).toContainReactComponent('label', {
+        className: 'Label VerticalAlign--Top',
```
Comment (chloerice) [inline src/components/OptionList/components/Option/tests/Option.test.tsx]:
```suggestion
        className: 'Label verticalAlignTop',
```

File: src/components/OptionList/components/Option/Option.tsx [hunk 8]
```diff
@@ -59,13 +63,15 @@ export function Option({
...
   );
 
   const multiSelectClassName = classNames(
     styles.Label,
     disabled && styles.disabled,
     active && styles.active,
     select && styles.select,
+    verticalAlign && styles[variationName('VerticalAlign--', verticalAlign)],
```
Comment (chloerice) [inline src/components/OptionList/components/Option/Option.tsx]:
```suggestion
    verticalAlign && styles[variationName('verticalAlign', verticalAlign)],
```

**Review by chloerice — APPROVED** [review]
Thank you so much for contributing, @chanonroy!! 🚀

File: src/components/OptionList/components/Option/Option.scss (hunk 1 above)
Comment (chanonroy) [reply src/components/OptionList/components/Option/Option.scss]: @chloerice Ah, amazing. Thanks for the clarification on this 😄

**Review by sarahill — APPROVED** [review]
Looks good! 👏 ✅ 🚢

**Pull request merged**